src/execution_impl.rs

```diff
@@ -83,21 +83,16 @@ pub(crate) fn exec(
     param: &str,
     interface: &dyn Interface,
 ) -> Result<Response> {
+    let mut env = Env::new(interface);
     let instance = match instance {
         Some(instance) => instance,
-        None => create_instance(limit, module, interface)?,
+        None => create_instance(limit, module, &env)?,
     };
-    let mut env = as_ffi_bindings::Env::default();
-    env.init(&instance)?;
+    env.init_with_instance(&instance)?;
```

Comment:
Is this necessary? I thought this would already be called internally by wasmer.

Reply:
Yes otherwise the env don't know the function in the next lines. I know they say they call it so I think there should be a way to make this code more usable by refactoring but for the moment it's needed.

Reply:
Ok I'd say give it a try by doing a `env.init_with_instance(&instance)?;` only in the `Some(instance)` case above. My guess is it's only called when we pass the env to the various things in `create_instance`.

Reply:
It don't work also. Because the run_function never pass an instance so it trigger the None one and in this case we need the init_instance.